pages/doc-components/CodeContainer.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import {LiveCode as MDXLive} from 'mdx-live'
+import copy from 'clipboard-copy'
+
+import {Relative, Absolute, Link, StyledOcticon} from '../..'
+import {Clippy, Check} from '@githubprimer/octicons-react'
+
+function CodeContainer({children: code, className: language}) {
+  if (language === 'language-.jsx') return <LiveCode initialCode={code} />
+  else return <StaticCode code={code} />
+}
+
+function LiveCode({initialCode}) {
+  const [code, setCode] = React.useState(initialCode)
+
+  return (
+    <Relative>
+      <MDXLive code={code} editorProps={{onChange: code => setCode(code)}} />
+      <CopyButton code={code} />
+    </Relative>
+  )
+}
+
+function StaticCode({code}) {
+  return (
+    <Relative>
+      <pre>{code}</pre>
+      <CopyButton code={code} />
+    </Relative>
+  )
+}
+
+/* Totally magic number based on what feels-right™ */
+const COPY_RESET_TIME = 1500
+
+function CopyButton({code}) {
+  const [copied, setCopied] = React.useState(false)
+
+  function onClick() {
+    copy(code)
+    setCopied(true)
+  }
+
+  /* Reset copied icon after some time */
+  React.useEffect(
+    function() {
+      setTimeout(function() {
+        if (copied) setCopied(false)
+      }, COPY_RESET_TIME)
+    },
+    [copied]
+  )
+
+  return (
+    <Absolute right={0} bottom={14}>
```

Comment:
`14px` is based on the height of one line of code with padding.

This makes it look good in both single and multiline code blocks:

<img width="145" alt="Screenshot 2019-05-23 at 8 41 39 PM" src="https://user-images.githubusercontent.com/1863771/58264466-8c5fc400-7d9b-11e9-8a51-97dae5224c43.png">